exporters/tests/test_in_memory_metric.py

```diff
@@ -0,0 +1,136 @@
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+#
+
+
+from unittest import mock
+
+import pytest
+from prometheus_client import REGISTRY
+from prometheus_client.registry import Collector
+
+from webhook.models.pelorus_webhook import CommitTimePelorusPayload, PelorusPayload
+
+# from pelorus.config import load_and_log
+from webhook.store.in_memory_metric import (
+    PelorusGaugeMetricFamily,
+    _pelorus_metric_to_dict,
+    pelorus_metric_to_prometheus,
+)
+
+metric_labels = list(_pelorus_metric_to_dict(CommitTimePelorusPayload).values())
+
+in_memory_test_committime_metrics = PelorusGaugeMetricFamily(
+    "test_committime_metrics",
+    "Test timestamp",
+    labels=metric_labels,
+)
+
+
+class CustomCommitCollector(Collector):
+    def __init_subclass__(cls) -> None:
+        super().__init_subclass__()
+
+        # make sure __hash__ is something prometheus' registry can handle properly.
+        cls.__hash__ = lambda self: id(self)  # type: ignore
+
+    def collect(self) -> PelorusGaugeMetricFamily:
+        yield in_memory_test_committime_metrics
+
+
+@pytest.mark.parametrize(
+    "name,timestamp,image_hash,namespace,commit_hash",
+    [
+        (
+            "todolist",
+            "timestamp_str",
+            "sha256:af4092ccbfa99a3ec1ea93058fe39b8ddfd8db1c7a18081db397c50a0b8ec77d",
+            "mynamespace",
+            "5379bad65a3f83853a75aabec9e0e43c75fd18fc",
+        ),
+    ],
+)
+def test_pelorus_gauge_metric_family(
+    name, timestamp, image_hash, namespace, commit_hash
+):
+    # custom_collector = load_and_log(CustomCommitCollector)
+    custom_collector = CustomCommitCollector()
+    metric_labels = list(_pelorus_metric_to_dict(CommitTimePelorusPayload).values())
+
+    commit_payload = CommitTimePelorusPayload(
+        app=name,
+        timestamp=timestamp,
+        image_sha=image_hash,
+        namespace=namespace,
+        commit_hash=commit_hash,
+    )
+
+    prometheus_commit_metric = pelorus_metric_to_prometheus(commit_payload)
+    in_memory_test_committime_metrics.add_metric(
+        commit_payload.commit_hash, prometheus_commit_metric, commit_payload.timestamp
+    )
+
+    REGISTRY.register(custom_collector)
+
+    metric_labels = {
+        "app": name,
+        "image_sha": image_hash,
+        "commit_hash": commit_hash,
+        "namespace": namespace,
+    }
+
+    query_result = REGISTRY.get_sample_value(
+        "test_committime_metrics",
+        labels=metric_labels,
+    )
+
+    assert query_result == timestamp
+
+
+def test_all_models_have_prometheus_mappings():
+    import webhook.models.pelorus_webhook as pelorus_webhook
```

Comment:
I would have all imports top level and use parametrize here, to do not have a for of asserts

Reply:
For this test I actually deliberately wanted to have this under the test, so we can see that we are testing *all* subclasses of PelorusPayload which are found in the `webhook.models.pelorus_webhook` module. This test is checking if all of the models have prometheus corresponding mappings.

However I see I forgot to update this test with comments, which I need to do before creating another revision of this PR.